src/main/scala/dx/core/io/DxWorkerPaths.scala

```diff
@@ -15,8 +41,59 @@ import wdlTools.util.Logger
   * files created with stdlib calls like "write_lines".
   * @param rootDir the root directory - typically the user's home directory
   */
-case class DxWorkerPaths(rootDir: Path)
-    extends ExecPaths(rootDir, rootDir.resolve(DxWorkerPaths.TempDir)) {
+case class DxWorkerPaths(rootDir: Path) extends BaseEvalPaths with ExecPaths {
+  def getRootDir(ensureExists: Boolean = false): Path = {
+    getOrCreateDir("root", rootDir, ensureExists)
+  }
+
+  def getTempDir(ensureExists: Boolean = false): Path = {
+    getOrCreateDir("temp", getRootDir(ensureExists).resolve(DxWorkerPaths.TempDir), ensureExists)
+  }
+
+  /**
+    * The execution directory - used as the base dir for relative paths (e.g. for glob search).
+    */
+  def getWorkDir(ensureExists: Boolean = false): Path = {
+    getOrCreateDir(DxWorkerPaths.WorkDir,
+                   getRootDir(ensureExists).resolve(DxWorkerPaths.WorkDir),
+                   ensureExists)
+  }
+
+  def getMetaDir(ensureExists: Boolean = false): Path = {
+    getOrCreateDir(DxWorkerPaths.MetaDir,
+                   getRootDir(ensureExists).resolve(DxWorkerPaths.MetaDir),
+                   ensureExists)
+  }
+
+  /**
+    * The file that has a copy of standard output.
+    */
+  def getStdoutFile(ensureParentExists: Boolean = false): Path = {
+    getMetaDir(ensureParentExists).resolve(DxWorkerPaths.StdoutFile)
+  }
+
+  /**
+    * The file that has a copy of standard error.
+    */
+  def getStderrFile(ensureParentExists: Boolean = false): Path = {
+    getMetaDir(ensureParentExists).resolve(DxWorkerPaths.StderrFile)
+  }
+
+  def getCommandFile(ensureParentExists: Boolean = false): Path = {
```

Comment:
Do we need them now if these metadirs are not used atm?

Reply:
They are used. It is where the job stdout, stderr, etc are stored.